index.js

```diff
@@ -397,6 +415,44 @@ async function fastifyStatic (fastify, opts) {
     const routeConfig = req.routeOptions?.config || req.routeConfig
     pumpSendToReply(req, reply, routeConfig.file, routeConfig.rootPath)
   }
+
+  async function generateHashes () {
+    fileHashes = new Map()
+
+    const roots = Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]
+    for (let rootPath of roots) {
+      rootPath = rootPath.split(path.win32.sep).join(path.posix.sep)
+      if (!rootPath.endsWith('/')) {
+        rootPath += '/'
+      }
+
+      const files = globSync('**/**', {
```

Comment:
I would recommend to avoid being sync here. Use an async. iterator, and start pushing things into fastq as soon as they come. So you can parallelize the globbing and hashing, given that the latter is CPU-bound, this will help.